src/Core/DecimalFunctions.h

```diff
@@ -121,32 +126,41 @@ DecimalComponents<typename DecimalType::NativeType> split(const DecimalType & de
  * If scale is to big, result is undefined.
  */
 template <typename DecimalType>
-typename DecimalType::NativeType getWholePart(const DecimalType & decimal, size_t scale)
+inline typename DecimalType::NativeType getWholePart(const DecimalType & decimal, size_t scale)
 {
     if (scale == 0)
         return decimal.value;
 
     return decimal.value / scaleMultiplier<typename DecimalType::NativeType>(scale);
 }
 
+
+template <typename DecimalType>
+inline typename DecimalType::NativeType getFractionalPartWithScaleMultiplier(
+        const DecimalType & decimal,
+        typename DecimalType::NativeType scale_multiplier)
+{
+    using T = typename DecimalType::NativeType;
+
+    T result = decimal.value;
+    if (result < T(0))
+        result *= T(-1);
```

Comment:
Maybe result = -result will look better?
(no difference in the behaviour indeed)